File: app/src/main/java/com/benoitquenaudon/tvfoot/red/app/injection/component/ActivityComponent.java
package com.benoitquenaudon.tvfoot.red.app.injection.component;

import com.benoitquenaudon.tvfoot.red.app.domain.main.MainActivity;
import com.benoitquenaudon.tvfoot.red.app.domain.match.MatchActivity;
import com.benoitquenaudon.tvfoot.red.app.domain.matches.MatchesActivity;
import com.benoitquenaudon.tvfoot.red.app.injection.module.ActivityModule;
import com.benoitquenaudon.tvfoot.red.app.injection.scope.ActivityScope;
import dagger.Subcomponent;

@ActivityScope @Subcomponent(modules = {
    ActivityModule.class, //
}) public interface ActivityComponent {
  void inject(MainActivity mainActivity);

  void inject(MatchesActivity matchesActivity);

  void inject(MatchActivity matchActivity);
}

File: app/src/main/java/com/benoitquenaudon/tvfoot/red/app/injection/component/ActivityComponent.kt
package com.benoitquenaudon.tvfoot.red.app.injection.component

import com.benoitquenaudon.tvfoot.red.app.domain.main.MainActivity
import com.benoitquenaudon.tvfoot.red.app.domain.match.MatchActivity
import com.benoitquenaudon.tvfoot.red.app.domain.matches.MatchesActivity
import com.benoitquenaudon.tvfoot.red.app.injection.module.ActivityModule
import com.benoitquenaudon.tvfoot.red.app.injection.scope.ActivityScope
import dagger.Subcomponent

@ActivityScope
@Subcomponent(modules = arrayOf(
    ActivityModule::class))
interface ActivityComponent {
  fun inject(mainActivity: MainActivity)

  fun inject(matchesActivity: MatchesActivity)

  fun inject(matchActivity: MatchActivity)
}